File: src/main/java/pg/eti/book/entity/impl/PublishingHouse.java
package pg.eti.book.entity.impl;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import pg.eti.literature.api.LiteraturePublisher;

import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PublishingHouse implements LiteraturePublisher {

	private String name;

	private int creationYear;

	private List<Book> books;

	@Override
	public String toString() {
		StringBuilder builder = new StringBuilder();
		builder.append(getPublisherInfo());
		return builder.toString();
		//builder.append()
	}

	@Override
	public String getPublisherInfo() {
		return String.format("PublishingHouse(name=%s, creationYear=%d)",
				name, creationYear);
	}
}

File: library-javascript/LibraryView/src/app/publishing-house/model/publishing-house.ts
export class PublishingHouse {
  id: String = '';
  name: String = '';
  creationYear: number = 0;

  constructor(id?: String, name?: String, creationYear?: number) {
    if (id) this.id = id;
    if (name) this.name = name;
    if (creationYear) this.creationYear = creationYear;
  }
}
